### PR TITLE
extraconfig based tether

### LINKED ISSUE #593: Migrate tether to pkg/vsphere/extraconfig
The tether is currently getting its configuration from guestinfo via a temporary hack - this should be switched out to the extraconfig package. This isn't just a drop in as it should also encompass moving the various scattered tracking structures (e.g. liveSessions) into the core configuration structure.

### PR DESCRIPTION
**Pull request by hickeng**
WIP: Moves the tether and portlayer create over onto the extraconfig serialization logic.

Currently untested as I'm failing to get vic-machine to work with my test env due to missing MAC addresses. I don't expect much to change however - only addressing of integration issues.

Fixes #593

### CODE REVIEW

**Comment by caglar10ur** [comment]
I get following when I try to create a container
```
time="2016-05-05T00:54:51Z" level=error msg="Operation failed: A specified parameter was not correct: config.extraConfig[\"common/notes\"] - (&{{} %!!(MISSING)s(*types.InvalidArgument=&{{{<nil> []}} config.extraConfig[\"common/notes\"]}) A specified parameter was not correct: config.extraConfig[\"common/notes\"]})"
```

File: cmd/tether/attach_test.go [hunk 1]
```diff
@@ -485,53 +463,46 @@ func TestAttachTwo(t *testing.T) {
...
 
+	// if there's no session command with guaranteed exit then tether needs to run in the background
+	sink := map[string]string{}
+	extraconfig.Encode(extraconfig.MapSink(sink), cfg)
+	src := extraconfig.MapSource(sink)
+	log.Debugf("Test configuration: %#v", sink)
+
 	go func() {
-		erR := run(cfg)
+		erR := run(src)
```
Comment (hmahmood) [inline cmd/tether/attach_test.go]: errr ... why erR?

**Comment by hmahmood** [comment]
lgtm

I think it would be easier if this goes in before my PR.

**Comment by caglar10ur** [comment]
We can merge https://github.com/vmware/vic/pull/643 and revert https://github.com/vmware/vic/pull/638/commits/cbe7d4f3ea990af542506c64a396b03c5b9a8a07

File: cmd/tether/tether_test.go [hunk 2]
```diff
@@ -316,6 +318,37 @@ func testTeardown(t *testing.T) {
...
+func startTether(t *testing.T, cfg *metadata.ExecutorConfig) extraconfig.DataSource {
+	store := map[string]string{}
+	sink := extraconfig.MapSink(store)
+	src := extraconfig.MapSource(store)
+	extraconfig.Encode(sink, cfg)
+	log.Debugf("Test configuration: %#v", sink)
+
+	// run the tether to service the attach
+	go func() {
+		erR := run(src, sink)
```
Comment (caglar10ur) [inline cmd/tether/tether_test.go]: what wrong with your err's :)

**Comment by caglar10ur** [comment]
LGTM :sheep: :it:

**Pull request merged**